File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.py
class Solution:
    def countNodes(self, root: Optional[TreeNode]) -> int:
        def cnt(node):
            if not node.left and not node.right:
                return

            nonlocal result
            if node.left:
                result += 1
                cnt(node.left)
            if node.right:
                result += 1
                cnt(node.right)


        if root is None: return 0
        result = 1
        cnt(root)
        return result

File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.java
class Solution {
    private int result = 0;

    public int countNodes(TreeNode root) {
        if (root == null) {
            return result;
        }
        result++;
        cnt(root);
        return result;
    }

    private void cnt(TreeNode node) {
        if (node.left == null && node.right == null) {
            return;
        }

        if (node.left != null) {
            result++;
            cnt(node.left);
        }

        if (node.right != null) {
            result++;
            cnt(node.right);
        }
    }
}